src/compiler/docs/readme/output-docs.ts

```diff
@@ -25,10 +43,20 @@ export const generateReadme = async (
   await Promise.all(
     readmeOutputs.map(async (readmeOutput) => {
       if (readmeOutput.dir) {
-        const readmeContent = generateMarkdown(userContent, docsData, cmps, readmeOutput);
-        const relPath = relative(config.srcDir, docsData.readmePath);
-        const absPath = join(readmeOutput.dir, relPath);
-        const results = await compilerCtx.fs.writeFile(absPath, readmeContent);
+        const relativeReadmePath = relative(config.srcDir, docsData.readmePath);
+        const readmeOutputPath = join(readmeOutput.dir, relativeReadmePath);
+
+        const currentReadmeContent =
+          readmeOutput.dir !== config.srcDir
+            ? // The user set a custom `.dir` property, which is where we're going
+              // to write the updated README. We need to read the non-automatically
+              // generated content from that file and preserve that.
+              await getUserReadmeContent(compilerCtx, readmeOutputPath)
```

Comment:
Would determining the README path at https://github.com/ionic-team/stencil/blob/b569ea7a9ad5d3b43eba3cf5c556cfab2c5e9e5f/src/compiler/docs/generate-doc-data.ts#L81 make it such that we don't need to read the README file from disk again?

Reply:
I think we can't really decide a 'canonical' README path at that point, because we're not operating on `docs-readme` output targets at that point

Or in other words, if I set a custom readme output dir on the `docs-readme` OT I don't think that should affect how the `docs-json` OT is handled

if we want that behavior then the readme location configuration should not be output target specific, but something in e.g. the `@Component` decorator

Reply:
> I think we can't really decide a 'canonical' README path at that point, because we're not operating on docs-readme output targets at that point

But hasn't a user specified the canonical README at that point (and we've defaulted to the directory of the component if they haven't)? If the `README.md` is the source of truth, doesn't that always affect the docs-json OT, even today?

Maybe the question here is, how many README files should exist, and which ones influence 'docs-json'? 

The Stencil component starter gives us a `README` by default for `my-component`, but on `main` today, that doesn't get regenerated/updated if we specify `dir` on the `'docs-readme'` OT. With that in mind, I'd expect the README in the custom dir to be the be the 'canonical' one (and where we pull README data from for 'docs-json').

For example:

1. `npm init stencil@latest component readme-out-test && cd $_ && npm i` to generate a component starter
2. Modifying `stencil.config.ts` before running a build:
```diff
diff --git a/stencil.config.ts b/stencil.config.ts
index 59cd15a..8167fb3 100644
--- a/stencil.config.ts
+++ b/stencil.config.ts
@@ -14,6 +14,7 @@ export const config: Config = {
     },
     {
       type: 'docs-readme',
+      dir: 'my-custom-dir'
     },
     {
       type: 'www',
```
3. Deleting `src/components/my-component/readme.md` that was a part of the starter
4. Running a build with `npm run build`

Would all result in the following dir tree:
```
my-custom-dir
└── components
    └── my-component
        └── readme.md
src
├── components
│   └── my-component
│       ├── my-component.css
│       ├── my-component.e2e.ts
│       ├── my-component.spec.ts
│       └── my-component.tsx
├── components.d.ts
├── index.html
├── index.ts
└── utils
    ├── utils.spec.ts
    └── utils.ts
```
Where `README.md` doesn't exist in `src`, because we don't generate it

Does that make sense? I've been staring at Jira for far too long today, and my brain feels a little static-y 😆 

As an aside, the behavior today on `main` and this branch creates a bit of a conundrum - we've left a README file next to the component in the filesystem, with no real way to clean it up ourselves (users will just have to do that). We have the same behavior on this branch, but I think that's out of scope here.

Reply:
> In particular, nothing prevents me from adding multiple different docs-readme output targets, each with different dir properties.

Ahhh that's what I was missing - I was only thinking about there being 0-1, and multiple READMEs scattered around the project directory as a result of multiple runs of `stencil build --docs` (and the like) with difference configurations.

I agree with your assessment there - thanks for the explanation!